### ISSUE TITLE
Allow to load (pre-trained) weights from checkpoint.

### ISSUE THREAD

**Issue by okuchaiev** [NVIDIA/OpenSeq2Seq#213]
1. Take pre-trained checkpoint from website and load weights only
2. Switch to another data-set and start training

**Comment by okuchaiev** [comment]
e.g. restore only trainable parameters from  checkpoint.
If trainable parameter is not in a checkpoint, initialize it randomly according to the initializer

**Comment by blisc** [comment]
It seems like we implemented this in #240. Can we close?

**Comment by ka-bu** [comment]
No, it shouldn't be closed. Doesn't work for me.

First of all, there are definitely some fixes needed:
- At [https://github.com/NVIDIA/OpenSeq2Seq/blob/master/open_seq2seq/utils/funcs.py#L121](url) the "fine_tuning" should be changed to " (base_ckpt_dir) or tf.train.latest_checkpoint(checkpoint_dir)" because otherwise the base_ckpt_dir (which will be set to the load_model path) is not used at all. 
- A minor bug: The 'log' folder should only be enforced enable_logs is set [https://github.com/NVIDIA/OpenSeq2Seq/blob/master/open_seq2seq/utils/utils.py#L639](url)

With these changes, I can actually load a model, according to the output (i.e. it lists the trainable variables that are restored). However, if I load a model and resume training, then the training losses are way to high. Seems that the training is starting again from scratch, even though the variables are loaded. Could it be that the assignments (loaded variables) are overwritten by some initialisation?

Btw "checkpoint_filename_with_path" parameter (and "load_fc") in [https://github.com/NVIDIA/OpenSeq2Seq/blob/4b953469dd574110679f74bd450b6465fff40a23/open_seq2seq/utils/helpers.py#L161](url) is never used, but it looks like this was supposed to be used for the load_model option.

**Comment by okuchaiev** [comment]
>> "However, if I load a model and resume training, then the training losses are way to high. Seems that the training is starting again from scratch, even though the variables are loaded."

- Could it be because of too high learning rate? E.g. When one trains from scratch the learning rate is typically decayed. Thus if pre-trained weights are loaded and training is restarted, if LR is as high as in the beginning, this could lead to divergence.

**Comment by ka-bu** [comment]
> Could it be because of too high learning rate? E.g. When one trains from scratch the learning rate is typically decayed. Thus if pre-trained weights are loaded and training is restarted, if LR is as high as in the beginning, this could lead to divergence.

Even if I set the learning rate to 0, the losses of the training batches seem to be way too high (evaluation loss seems to match, though). 

The "load_model" parameter does definitely not do anything right now.  And I'm not sure that the fix I mentioned above totally solves the problem (though the debugging output tells me that the assign ops are executed in the restore_certain_variables). 

Would be great if one of you could test if this fix works (and I overlooked something in my test setup), or if the loaded parameters are overwritten.

**Comment by ka-bu** [comment]
The discrepancy in the train errors must be due to something else that I overlook.

I tested the fix by first loading a model and setting save_checkpoint_steps to 0, and then comparing the weights from the checkpoint of step 1 to the weights of the loaded model. Works fine :) . 

@okuchaiev PR to master?

**Comment by okuchaiev** [comment]
@ka-bu yes, please send a PR to master. Thank you!
